dash-pipeline/bmv2/stages/tunnel1.p4

```diff
@@ -0,0 +1,41 @@
+#ifndef _DASH_STAGE_TUNNEL1_P4_
+#define _DASH_STAGE_TUNNEL1_P4_
+
+control tunnel1_stage(
+    inout headers_t hdr,
+    inout metadata_t meta)
+{
+    action set_tunnel1_attrs(
+            @SaiVal[type="sai_ip_address_t"]
+            IPv4Address underlay_dip,
+            @SaiVal[type="sai_dash_encapsulation_t", default_value="SAI_DASH_ENCAPSULATION_VXLAN"]
+            dash_encapsulation_t dash_encapsulation,
+            bit<24> tunnel_key) {
+    push_action_static_encap(hdr = hdr,
```

Comment:
Would this overwrite the push_action_static_encap action performed in set_private_link_mapping action?

Reply:
and once this push_action is moved into each routing type, we can move the tunnel table into the action apply stage.

Reply:
and the set tunnel1 attr can be simply push the action and update the tunnel id in the metadata. everything else should be done in the later tunnel action, otherwise the 2 layer of encap won't work, as we only save 1 single layer information in our metadata.

Reply:
talking about the bug, I wonder if we could add a test for it : D

Reply:
Moved the tunnel stage after routing_action_apply. Working on a test, will update with a dedicated commit to this PR.